persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/interceptors/ProxyProtocolQueryInterceptor.java

```diff
@@ -71,11 +71,13 @@
 
   private final Resolver resolver;
   private final String proxyDnsName;
+  private final String internalProxyDnsName;
   private final int proxyPort;
   private final long resolveDelaySecs;
   private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
   private final Map<InetAddress, Set<String>> tokensCache = new ConcurrentHashMap<>();
-  private volatile Set<InetAddress> peers = Collections.emptySet();
+  private volatile Set<InetAddress> externalPeers = Collections.emptySet();
+  private volatile Set<InetAddress> internalPeers = Collections.emptySet();
```

Comment:
As far as I can tell, this interceptor is shared across connections (it's set on `ClientState.cqlQueryHandler`, which is static).

We can't predict if connections are going to be internal or public (or both), so we have to handle both cases.